climpred/classes.py

```diff
@@ -645,10 +649,15 @@ def verify(
             results for the initialized ensemble (``init``) and any reference forecasts
             verified.
         """
-        has_dataset(self._datasets["control"], "control", "compute a metric")
+        # pass fake control if no control dataset available
+        # compute_perfect_model does not use control anyways
```

Comment:
I still don't understand this. Why does `compute_perfect_model` take `control` as an argument if it doesn't use it? Why do the comparisons then assume that the control member is 0? That seems restrictive. If a user adds a control to the `PM` object, they assume that `m2c` is comparing to that control run. But in reality it's completely ignoring that object and instead comparing to `initialized.isel(member=0)`. Which is counterintuitive. I know with PM, any member can be the control, but the user is probably assuming it's getting compared to the control in their object.

Reply:
> I still don't understand this. Why does compute_perfect_model take control as an argument if it doesn't use it? 

Legacy. in the beginning I had `compute_perfect_model(ds, control)` and `compute_persistence(ds, control)` then I figured out I dont really need control for `compute_perfect_model(ds, control)`. I could actually pass `control` as `metric_kwargs` to do the normalization for `nmse` but I just use the `initialized.std(['lead','init','member'])` which ends up pretty much the same.

Reply:
> Why do the comparisons then assume that the control member is 0? That seems restrictive.

In a perfect-model framework there is no difference between member 0 and member 1. its arbitrary labeling/naming.

Reply:
> If a user adds a control to the PM object, they assume that m2c is comparing to that control run. But in reality it's completely ignoring that object and instead comparing to initialized.isel(member=0)

I should make this more clean in the docs of `m2c` then.

Reply:
m2c: `Compare all other members forecasts to control member verification.`

before: https://github.com/aaronspring/climpred/blob/ce9ab1c05b2375a5dafe6cb9e51234ac66873f0f/climpred/comparisons.py#L165

Reply:
now hopefully better to understand

Reply:
@aaronspring 

1. Can you also clean up the `compute_perfect_model` in `prediction.py`? It takes `control` and documents it but then doesn't use it.

2. Have you looked at the variability in metrics/scores based on comparing ensemble mean to member=0, member=1, member=2, and so on? I get that any member can be viewed as a verification, but to always select their member=0 seems problematic potentially.

Reply:
2. This is only the case for m2c and e2c.

Reply:
1. will adapt

Reply:
1. I will not adapt this. will be refactored at some point. I will document the non-use of control. The problem with cleanup is: inside `bootstrap_compute` (wrapping `bootstrap_pm` and `bootstrap_hindcast`) I rely on `compute_hindcast` and `compute_perfect_model` sharing the same inputs: 1. `initialized` 2. `control/verif`